File: benchmarks/nodemanual/batch1.js
require('./nodemanual.js')();

async function main() {
    const base_dir = "../../../datasets/";
    const [warmup_query, repeat_query] = init(base_dir);

    const twitter_small = "twitter_small_records.json";
    const walmart_small = "walmart_small_records.json";
    const bestbuy_small = "bestbuy_small_records.json";

    await execute(warmup_query, repeat_query, twitter_small, query_TT1, "TT1");
    await execute(warmup_query, repeat_query, twitter_small, query_TT2, "TT2");
    await execute(warmup_query, repeat_query, twitter_small, query_TT3, "TT3");
    await execute(warmup_query, repeat_query, twitter_small, query_TT4, "TT4");

    await execute(warmup_query, repeat_query, walmart_small, query_WM, "WM");

    await execute(warmup_query, repeat_query, bestbuy_small, query_BB, "BB");
}

main()

File: benchmarks/rapidjson/batch1.cpp
#include "rapidjson.cpp"

int main(int argc, char *argv[]) {
    char base_dir[] = "../../../datasets/";
    init(argc, argv, base_dir);

    char twitter_small[50] = "twitter_small_records.json";
    char walmart_small[50] = "walmart_small_records.json";
    char bestbuy_small[50] = "bestbuy_small_records.json";

    execute(twitter_small, query_TT1, "TT1");
    execute(twitter_small, query_TT2, "TT2");
    execute(twitter_small, query_TT3, "TT3");
    execute(twitter_small, query_TT4, "TT4");  

    execute(walmart_small, query_WM, "WM");

    execute(bestbuy_small, query_BB, "BB");

    return 0;
}